packages/bitcoin-dlc-provider/lib/BitcoinDlcProvider.ts

```diff
@@ -371,18 +433,45 @@ export default class BitcoinDlcProvider
     totalCollateral: bigint,
   ) {
     switch (contractDescriptor.type) {
-      case MessageType.ContractDescriptorV0:
+      case MessageType.ContractDescriptorV0: {
         throw Error('ContractDescriptorV0 not yet supported');
+      }
       case MessageType.ContractDescriptorV1:
-        // eslint-disable-next-line no-case-declarations
-        return this.GetPayoutsFromPayoutFunction(
-          dlcOffer,
-          contractDescriptor as ContractDescriptorV1,
-          oracleInfo,
-          totalCollateral,
-        );
-      default:
+        {
+          const contractDescriptorV1 = contractDescriptor as ContractDescriptorV1;
+          const payoutFunction = contractDescriptorV1.payoutFunction as PayoutFunctionV0;
+
+          // TODO: add a better check for this
+          const payoutCurvePiece = payoutFunction.pieces[0].payoutCurvePiece;
```

Comment:
What did you have in mind for a better check here?

Reply:
Right now I only check the type of the `first` piece out of `n` pieces. If we have a curve in the future made of multiple different polynomials + hyperbolas, then this would not work.